src/main/java/org/spongepowered/common/data/processor/value/entity/BoatTreeTypeValueProcessor.java

```diff
@@ -0,0 +1,110 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.processor.value.entity;
+
+import net.minecraft.entity.item.EntityBoat;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.block.ImmutableTreeData;
+import org.spongepowered.api.data.manipulator.mutable.block.TreeData;
+import org.spongepowered.api.data.type.TreeType;
+import org.spongepowered.api.data.type.TreeTypes;
+import org.spongepowered.api.data.value.ValueContainer;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.api.data.value.mutable.Value;
+import org.spongepowered.common.data.manipulator.mutable.block.SpongeTreeData;
+import org.spongepowered.common.data.processor.common.AbstractEntitySingleDataProcessor;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeValue;
+import org.spongepowered.common.data.value.mutable.SpongeValue;
+
+import java.util.Optional;
+
+public class BoatTreeTypeValueProcessor extends AbstractEntitySingleDataProcessor<EntityBoat, TreeType, Value<TreeType>, TreeData, ImmutableTreeData> {
+
+    public BoatTreeTypeValueProcessor() {
+        super(EntityBoat.class, Keys.TREE_TYPE);
+    }
+
+    @Override
+    protected TreeData createManipulator() {
+        return new SpongeTreeData();
+    }
+
+    @Override
+    protected boolean set(EntityBoat dataHolder, TreeType value) {
+        if (value == TreeTypes.OAK) {
+            dataHolder.setBoatType(EntityBoat.Type.OAK);
+        } else if (value == TreeTypes.SPRUCE) {
+            dataHolder.setBoatType(EntityBoat.Type.SPRUCE);
+        } else if (value == TreeTypes.JUNGLE) {
+            dataHolder.setBoatType(EntityBoat.Type.JUNGLE);
+        } else if (value == TreeTypes.DARK_OAK) {
+            dataHolder.setBoatType(EntityBoat.Type.DARK_OAK);
+        } else if (value == TreeTypes.BIRCH) {
+            dataHolder.setBoatType(EntityBoat.Type.BIRCH);
+        } else if (value == TreeTypes.ACACIA) {
+            dataHolder.setBoatType(EntityBoat.Type.ACACIA);
+        } else {
+            return false;
+        }
+
+        return true;
+    }
+
+    @Override
+    protected Optional<TreeType> getVal(EntityBoat dataHolder) {
+        switch (dataHolder.getBoatType()) {
+            case OAK:
+                return Optional.of(TreeTypes.OAK);
+            case SPRUCE:
+                return Optional.of(TreeTypes.SPRUCE);
+            case BIRCH:
+                return Optional.of(TreeTypes.BIRCH);
+            case JUNGLE:
+                return Optional.of(TreeTypes.JUNGLE);
+            case ACACIA:
+                return Optional.of(TreeTypes.ACACIA);
+            case DARK_OAK:
+                return Optional.of(TreeTypes.DARK_OAK);
+        }
+
+        return Optional.empty();
```

Comment:
I'm not sure about this one, maybe it would be better if we throw an IAE here to faster detect new enum values. Or at lest use some logging.

Reply:
So my issue with an IllegalArgumentException, is that if a mod adds a new type of tree the mapping won't be here.